esp/src/src-react/components/SourceEditor.tsx

```diff
@@ -11,11 +11,13 @@ import { ShortVerticalDivider } from "./Common";
 interface SourceEditorProps {
     text: string;
     readonly?: boolean;
+    mode: "xml" | "text";
 }
 
-export const XMLSourceEditor: React.FunctionComponent<SourceEditorProps> = ({
+const SourceEditor: React.FunctionComponent<SourceEditorProps> = ({
     text = "",
-    readonly = false
+    readonly = false,
+    mode = text
```

Comment:
This is probably something I don't yet understand about TypeScript. But should this be "text"? It looks like it's initializing to the value of the destructured property named "text".